src/core/observer/watcher.js

```diff
@@ -80,13 +80,21 @@ export default class Watcher {
     } else {
       this.getter = parsePath(expOrFn)
       if (!this.getter) {
+        if (this.getter === false) {
```

Comment:
I don't like that we are relying on magic return values (`false`) here, as the logic is separated in two places. Also the check is warning only so it's better to place it right here inside a `if (process.env.NODE_ENV !== 'production') { ... }` block, using an inline regexp.

Reply:
@yyx990803 Thanks for taking the time to review this PR. 

I've gone ahead and made the commits to stop relying on random return values. I also tried your suggestion with having the check in an `if (process.env.NODE_ENV !== 'production') { ... }` block, but felt that it was unclear to have two regex checks in different places (`watcher` and `parsePath`) and did not resolve your comment on the logic being separated, so I grouped them together.

Thanks again and please let me know what you think!